tracetools_trace/tracetools_trace/trace.py

```diff
@@ -307,7 +377,12 @@ def resume(args: argparse.Namespace) -> int:
     :return: the return code (0 if successful, 1 otherwise)
     """
     def work() -> int:
-        lttng.lttng_start(session_name=args.session_name)
+        session_name = args.session_name
+        if args.dual_session:
+            snapshot_session_name = session_name + path.SNAPSHOT_SESSION_SUFFIX
+            lttng.lttng_record_snapshot(session_name=snapshot_session_name)
+            session_name += path.RUNTIME_SESSION_SUFFIX
+        lttng.lttng_start(session_name=session_name)
```

Comment:
Interesting, I didn't think about triggering the snapshot when resuming the normal tracing session too! I think this would be worth a quick comment to point it out:

```suggestion
        session_name = args.session_name
        if args.dual_session:
            # Trigger snapshot to record any init events missed while paused
            snapshot_session_name = session_name + path.SNAPSHOT_SESSION_SUFFIX
            lttng.lttng_record_snapshot(session_name=snapshot_session_name)
            session_name += path.RUNTIME_SESSION_SUFFIX
        lttng.lttng_start(session_name=session_name)
```